### ISSUE TITLE
`TypeError: __init__() got an unexpected keyword argument 'path'`

### ISSUE THREAD

**Issue by homeworkprod** [rq/rq#1398]
Versions:
```
hiredis==1.1.0
redis==3.5.3
rq==1.7.0
```

Launch script is basically:
```py
with connection():
    queues = [get_queue(app)]

    worker = Worker(queues)
    worker.work(with_scheduler=True)
```

Resulting in this exception right on start of the worker:
```pytb
TypeError: __init__() got an unexpected keyword argument 'path'
  File "worker.py", line 32, in <module>
    worker.work(with_scheduler=True)
  File "rq/worker.py", line 546, in work
    self.scheduler.acquire_locks()
  File "rq/scheduler.py", line 95, in acquire_locks
    if self.connection.set(self.get_locking_key(name), pid, nx=True, ex=60):
  File "rq/scheduler.py", line 69, in connection
    self._connection = self._connection_class(**self._connection_kwargs)
```

Observations:
- This works locally with both rq v1.6.1 and v1.7.0.
- This fails on prod with both rq v1.6.1 and v1.7.0.
- It works without the `with_scheduler=True` keyword argument.
- Locally, the Redis URL is a network socket (`redis://127.0.0.1:6379/0`), but on prod it's a file socket (`unix:///var/run/redis/redis.sock?db=0`).

That's all I have for now.

**Comment by homeworkprod** [comment]
Further investigation results:
- Python version is 3.7.3 both locally and on prod.
- rq passes a connection object around which is an instance of `redis.client.Redis` from the `redis` library.
- Class `redis.client.Redis` has a field `unix_socket_path` which is [put into its connection keyword arguments with the key `path`])https://github.com/andymccurdy/redis-py/blob/master/redis/client.py#L776).
- rq's worker [creates a scheduler instance and passes the connection to it](https://github.com/rq/rq/blob/v1.7.0/rq/worker.py#L544).
- When the scheduler [creates its own connection object for the first time](https://github.com/rq/rq/blob/v1.7.0/rq/scheduler.py#L69), the exception is raised as `redis.client.Redis`'s constructor accepts only `unix_socket_path` as a keyword argument but not `path`.

Sooo … yeah. I'm not convinved that changing keyword names is the cleanest strategy here.

**Comment by homeworkprod** [comment]
And it probably makes sense to add a test case that combines a Redis socket path and (a worker with) a scheduler, as it looks like this hasn't been tested before releasing the scheduler mechanism (not blaming anyone!).

**Comment by homeworkprod** [comment]
OK, so as far as I understand it, the Redis library adjusts the keyword name to pass it to `UnixDomainSocketConnection`, which is also set to the key `connection_class`.

What rq's scheduler implementation does is:
- [Pull the rewritten keyword arguments from the connection pool](https://github.com/rq/rq/blob/v1.7.0/rq/scheduler.py#L43)
- [Keeps the Redis client instance around as `self._connection_class`](https://github.com/rq/rq/blob/v1.7.0/rq/scheduler.py#L47)
- [Correctly pull the connection class from the connection pool](https://github.com/rq/rq/blob/v1.7.0/rq/scheduler.py#L48)
- [Try to create a new connection from the Redis client](https://github.com/rq/rq/blob/v1.7.0/rq/scheduler.py#L69)

Creating the new connection works when using the fitting class from `self.connection_class`. Later, however, it fails when `set` should be called on it.

So we do have an actual `redis.connection.Connection`, but want to use it like a `redis.client.Redis` instance later. I didn't find a clean way to create a `Redis` instance from an existing connection, though.

So far it *seems* a bit that rq a bit loosely interchanges `Redis` client instances and *actual* `redis.connection.Connection` (subclass) instances.

The API of the redis client library seems to do its part as well: for a non-SSL TCP connection the keywords are passed from `Redis`' constructor to the connection pool constructor, but both UNIX domain socket as well as SSL connections need to spe a different class.

Not sure how this can be tackled.
- Maybe separate `redis.client.Redis` and `redis.connection.Connection` better?
- Maybe see if the `from_url` factory methods are a viable alternative?
- Find a way to extract the original keyword arguments passed to `Redis` from the connection pool.
- Do what feels like a hack and rewrite `path` to `unix_socket_path`?
- Talk to the `redis` library folks and suggest a way to create `Redis` from an existing connection, or some other approach?

**Comment by lowercase00** [comment]
Closing this as it seems to be fixed by #1403. Feel free to reopen if there are any other issues.
